DataFormats/TrackReco/interface/trackAlgoPriorityOrder.h

```diff
@@ -0,0 +1,155 @@
+#ifndef DataFormats_TrackReco_trackAlgoPriorityOrder_h
+#define DataFormats_TrackReco_trackAlgoPriorityOrder_h
+
+#include "DataFormats/TrackReco/interface/TrackBase.h"
+
+#include <array>
+
+/**
+ * The trackAlgoPriorityOrder maps an reco::TrackBase::TrackAlgorithm
+ * enumerator to its priority in track list merging. The mapping is
+ * needed because the order of the enumerators themselves does not, in
+ * general, convey useful information.
+ */
+namespace impl {
+
+  /**
+   * This array defines the priority order in merging for the
+   * algorithms. The order is ascending, i.e. the first algorithm has
+   * the highest priority etc. The size of the array should be
+   * reco::TrackBase:algoSize (checked below with static_assert), and
+   * each reco::TrackBase::TrackAlgorithm enumerator should be in the
+   * array exactly once (checked below in findIndex() function).
+   */
+  constexpr reco::TrackBase::TrackAlgorithm algoPriorityOrder[] ={
+    reco::TrackBase::undefAlgorithm,
+    reco::TrackBase::ctf,
+    reco::TrackBase::rs,
+    reco::TrackBase::cosmics,
+    reco::TrackBase::initialStep,
+    reco::TrackBase::detachedTripletStep,
```

Comment:
due to the copy of the file it is not obvious to spot that "detachedTripletStep" has been moved to its proper place.
This will generate some regression as now detachedTripletStep will be preferred above lowPtTripletStep and pixelPairStep if everything else is "equal"